From e0540030b8ccd231e2da2bb62331715886498dae Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Fri, 18 Apr 2025 18:07:15 -0500
Subject: [PATCH 2/6] BLD: Find nvimgcodec package instead of downloading
 source

---
 cmake/Dependencies.common.cmake | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/cmake/Dependencies.common.cmake b/cmake/Dependencies.common.cmake
index 8f6cf22a..6568bc85 100644
--- a/cmake/Dependencies.common.cmake
+++ b/cmake/Dependencies.common.cmake
@@ -285,18 +285,7 @@ if(BUILD_NVIMAGECODEC)
     endif()
 
     # Note: We are getting the x86_64 tarball, but we are only interested in the headers.
-    include(FetchContent)
-    FetchContent_Declare(
-      nvimgcodec_headers
-      URL      https://developer.download.nvidia.com/compute/nvimgcodec/redist/nvimgcodec/linux-x86_64/nvimgcodec-linux-x86_64-0.5.0.13-archive.tar.xz
-      URL_HASH SHA512=f220f06315e18dece601971c0b31798cc819522ed0daf651fcc12e5436f62e051de8e7171a11e8e10af25930493b00c2e3a214a0e1eabb27ab57748b9966d3bd
-    )
-    FetchContent_Populate(nvimgcodec_headers)
-    set(nvimgcodec_SEARCH_PATH "${nvimgcodec_headers_SOURCE_DIR}/${CUDA_VERSION_MAJOR}/include")
-    find_path(nvimgcodec_INCLUDE_DIR nvimgcodec.h PATHS "${nvimgcodec_SEARCH_PATH}")
-    if (${nvimgcodec_INCLUDE_DIR} STREQUAL "nvimgcodec_INCLUDE_DIR-NOTFOUND")
-      message(FATAL_ERROR "nvimgcodec not found in ${nvimgcodec_SEARCH_PATH} - something went wrong with the download")
-    endif()
+    find_package(nvimgcodec ${NVIMGCODEC_MIN_VERSION}...<${NVIMGCODEC_MAX_VERSION} REQUIRED)
     message(STATUS "Using nvimgcodec_INCLUDE_DIR=${nvimgcodec_INCLUDE_DIR}")
     include_directories(SYSTEM ${nvimgcodec_INCLUDE_DIR})
 
-- 
2.49.0

