From 61ae714bd7dc47f0cc8244c02e4251f9ec851697 Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Fri, 18 Apr 2025 18:08:01 -0500
Subject: [PATCH 3/3] BLD: Allow use of nvimgcodec v0.5

---
 cmake/Dependencies.common.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Dependencies.common.cmake b/cmake/Dependencies.common.cmake
index 4448bf03..b73cee47 100644
--- a/cmake/Dependencies.common.cmake
+++ b/cmake/Dependencies.common.cmake
@@ -282,7 +282,7 @@ if(BUILD_NVIMAGECODEC)
   set(NVTIFF_MAX_VERSION "0.5.0")
 
   set(NVIMGCODEC_MIN_VERSION "0.4.1")
-  set(NVIMGCODEC_MAX_VERSION "0.5.0")
+  set(NVIMGCODEC_MAX_VERSION "0.6.0")
   message(STATUS "nvImageCodec - requires version >=${NVIMGCODEC_MIN_VERSION}, <${NVIMGCODEC_MAX_VERSION}")
   if (WITH_DYNAMIC_NVIMGCODEC)
     message(STATUS "nvImageCodec - dynamic load")
-- 
2.34.1

