diff --git a/src/cvcuda/priv/CMakeLists.txt b/src/cvcuda/priv/CMakeLists.txt
index deac31c2..581c1d70 100644
--- a/src/cvcuda/priv/CMakeLists.txt
+++ b/src/cvcuda/priv/CMakeLists.txt
@@ -95,6 +95,17 @@ add_library(cvcuda_priv STATIC
     ${CV_CUDA_PRIV_FILES}
 )

+set(CV_CUDA_PRIV_LINK_LIBS "")
+if ("OpFindHomography.cu" IN_LIST CV_CUDA_PRIV_FILES)
+    list(APPEND CV_CUDA_PRIV_LINK_LIBS
+        CUDA::cudart_static
+        CUDA::cusolver_static
+        CUDA::cublas_static
+        CUDA::cublasLt_static
+    )
+endif()
+
+
 target_link_libraries(cvcuda_priv
     PUBLIC
         nvcv_types
@@ -102,9 +113,6 @@ target_link_libraries(cvcuda_priv
         cvcuda_headers
         nvcv_util_sanitizer
         cvcuda_legacy
-        CUDA::cudart_static
-           CUDA::cusolver_static
-        CUDA::cublas_static
-        CUDA::cublasLt_static
+        ${CV_CUDA_PRIV_LINK_LIBS}
         -lrt
 )
